app/src/main/java/com/appculinaryrecipes/fragments/FavouritesFragment.java

```diff
@@ -0,0 +1,175 @@
+        View view = fragmentFavouritesBinding.getRoot();
+        recyclerView = fragmentFavouritesBinding.favouritesRecyclerView;
+        swipeRefreshLayout = fragmentFavouritesBinding.favouritesSwipeRefreshLayout;
+        recyclerView.setHasFixedSize(true);
+
+        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this.getContext());
+        recyclerView.setLayoutManager(linearLayoutManager);
+
+        recyclerViewAdapter = new RecyclerViewAdapter(this.getContext());
+        recyclerView.setAdapter(recyclerViewAdapter);
+        leadData();
+        recyclerView.addOnScrollListener(new RecyclerView.OnScrollListener() {
+            @Override
+            public void onScrolled(@NonNull RecyclerView recyclerView, int dx, int dy) {
+                LinearLayoutManager layoutManager = (LinearLayoutManager) recyclerView.getLayoutManager();
+                int itemCount = layoutManager.getItemCount();
+                int lastCompletelyVisibleItemPosition = layoutManager.findLastCompletelyVisibleItemPosition();
+                if (itemCount < lastCompletelyVisibleItemPosition + 3) {
+                    if (!isLoading) {
+                        isLoading = true;
+                        leadData();
+                    }
+                }
+
+            }
+        });
+
+
+        return view;
+    }
+
+    private void leadData() {
+        swipeRefreshLayout.setRefreshing(true);
+        String userUID = getUser();
+        ArrayList<Recipe> recipeArrayList = new ArrayList<>();
+
+        FirebaseFirestore db = FirebaseFirestore.getInstance();
+        DocumentReference documentReference = db.collection("favourites").document(userUID);
+        Query query;
+
+        documentReference.get().addOnCompleteListener(new OnCompleteListener<DocumentSnapshot>() {
+            @Override
+            public void onComplete(@NonNull Task<DocumentSnapshot> task) {
+                if (task.isSuccessful()) {
+                    DocumentSnapshot doc = task.getResult();
+                    if (doc.exists()) {
+                        favourites = (ArrayList<String>) doc.get("favourites");
+                    }
+                }
+            }
+        });
+
+        if (lastResult == null) {
+            query = db
+                    .collection(DB_RECIPE_DOCUMENT_NAME)
+                    .limit(PAGE_ITEM_SIZE);
+        } else {
+            query = db
+                    .collection(DB_RECIPE_DOCUMENT_NAME)
+                    .startAfter(lastResult)
+                    .limit(PAGE_ITEM_SIZE);
+        }
+
+
+        query
+                .get()
+                .addOnCompleteListener(task -> {
+                    if (task.isSuccessful()) {
+                        for (QueryDocumentSnapshot document : task.getResult()) {
+                            Map<String, Object> data = document.getData();
+                            String id = document.getId();
+                            if (favourites.contains(id)) {
```

Comment:
Tutaj dopiero to sprawdzasz. Niby może być, ale ciągniesz tylko PAGE_ITEM_SIZE przepisów najpierw i potem dociągasz inne, ale w tych 8 może nie być żadnego z ulubionych, wiec bez sensu troche. Lepiej wyciągnąć tylko ulubione i do nich paginacje dorobić